assets/js/modules/subscribe-with-google/datastore/settings.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * `modules/subscribe-with-google` data store: settings.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { TYPE_MODULES } from '../../../components/data/constants';
+import { invalidateCacheGroup } from '../../../components/data/invalidate-cache-group';
+import { STORE_NAME } from './constants';
+import { INVARIANT_DOING_SUBMIT_CHANGES, INVARIANT_SETTINGS_NOT_CHANGED } from '../../../googlesitekit/data/create-settings-store';
+import { createStrictSelect } from '../../../googlesitekit/data/utils';
+import { isValidProducts, isValidPublicationID } from '../util/validation';
+
+// Invariant error messages.
+export const INVARIANT_INVALID_PUBLICATION_ID = 'a valid publicationID is required';
+export const INVARIANT_INVALID_PRODUCTS = 'a valid products string is required';
+
+export async function submitChanges( { select, dispatch } ) {
+	// This action shouldn't be called if settings haven't changed,
+	// but this prevents errors in tests.
+	if ( select( STORE_NAME ).haveSettingsChanged() ) {
+		const { error } = await dispatch( STORE_NAME ).saveSettings();
+
+		if ( error ) {
+			return { error };
+		}
+	}
+
+	await API.invalidateCache( 'modules', 'subscribe-with-google' );
+	// TODO: Remove once legacy dataAPI is no longer used.
+	invalidateCacheGroup( TYPE_MODULES, 'subscribe-with-google' );
+
+	return {};
+}
+
+export function validateCanSubmitChanges( select ) {
```

Comment:
Related to my above comment: This function is what will internally power the `canSubmitChanges` selector. So as long as you have something here that ensures it's not possible to submit until both `publicationID` and `products` are not empty, you can use the `canSubmitChanges` selector to get similar behavior to what you intended above in `PrototypeForm`, plus all the other scenarios where that button should be disabled - with the benefit of it being handled in a central place.